tests/runtime-tests/src/main.rs

```diff
@@ -0,0 +1,304 @@
+use std::{
+    io::{Read, Write},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use anyhow::Context;
+
+fn main() -> anyhow::Result<()> {
```

Comment:
A test harness like this would be useful for user app testing too: https://github.com/fermyon/spin/issues/2063